ingestion/src/metadata/ingestion/source/dashboard/sigma/metadata.py

```diff
@@ -0,0 +1,228 @@
+        for chart in charts or []:
+            try:
+                if filter_by_chart(self.source_config.chartFilterPattern, chart.name):
+                    self.status.filter(chart.name, "Chart Pattern not allowed")
+                    continue
+                yield Either(
+                    right=CreateChartRequest(
+                        name=EntityName(str(chart.elementId)),
+                        displayName=chart.name,
+                        chartType=get_standard_chart_type(chart.vizualizationType),
+                        service=FullyQualifiedEntityName(
+                            self.context.get().dashboard_service
+                        ),
+                        sourceUrl=SourceUrl(dashboard_details.url),
+                        description=Markdown(dashboard_details.description),
+                    )
+                )
+            except Exception as exc:
+                yield Either(
+                    left=StackTraceError(
+                        name="Chart",
+                        error=(
+                            "Error to yield dashboard chart for : "
+                            f"{chart.elementId} and {dashboard_details}: {exc}"
+                        ),
+                        stackTrace=traceback.format_exc(),
+                    )
+                )
+
+    def yield_dashboard_lineage_details(
+        self, dashboard_details: WorkbookDetails, db_service_name: Optional[str]
+    ):
+        """
+        yield dashboard lineage
+        """
+        if not db_service_name:
+            return
+        to_fqn = fqn.build(
+            self.metadata,
+            entity_type=LineageDashboard,
+            service_name=self.config.serviceName,
+            dashboard_name=str(dashboard_details.workbookId),
+        )
+        to_entity = self.metadata.get_by_name(
+            entity=LineageDashboard,
+            fqn=to_fqn,
+        )
+        for chart in self.context.get().charts or []:
+            nodes = self.client.get_lineage_details(dashboard_details.workbookId, chart)
+            for node in nodes:
+                if node.node_schema:
+                    try:
+                        from_fqn = fqn.build(
+                            self.metadata,
+                            entity_type=Table,
+                            service_name=db_service_name,
+                            schema_name=node.node_schema,
+                            table_name=node.name,
+                            database_name="",
+                        )
+                        from_entity = self.metadata.get_by_name(
+                            entity=Table,
+                            fqn=from_fqn,
+                        )
+                        if from_entity and to_entity:
+                            yield self._get_add_lineage_request(
+                                to_entity=to_entity, from_entity=from_entity
```

Comment:
any way we can get column level lineage? Also, does sigma have the concept of DashboardDataModels? Or dashboards are directly connected to the source data

Reply:
sigma does have a concept of DashboardDataModels but it is introducted recently and its still in beta.
currently dashboards are directly connected to source data in sigma.
DashboardDataModels as in beta in sigma does not have any apis for them currently to fetch metadata from.

Reply:
ok got it, let's put it in a comment in the method as well as in the documentation. Any way we can get col-level lineage though?

Reply:
ill check on the column level lineage in the sigma APIs

Reply:
As of now we can build column level lineage between dashboard data models and the table